ferrilata-devilline/Controllers/BadgesController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using ferrilata_devilline.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace ferrilata_devilline.Controllers
+{
+    public class BadgesController : Controller
+    {
+        [HttpGet]
+        [Route("/api/badges")]
+        public IActionResult getGadgets()
+        {
+            var request = Request;
```

Comment:
You could use `Request.Headers` without putting it in a variable, I mean you are just cheking it in the if statement, it would be a little longer but in that case put the 2nd expression in a new line after the `&&`